deepspeed/runtime/utils.py

```diff
@@ -266,7 +266,8 @@ def _has_inf_or_nan(x, i):
             # if x is half, the .float() incurs an additional deep copy, but it's necessary if
             # Pytorch's .sum() creates a one-element tensor of the same type as x
             # (which is true for some recent version of pytorch).
-            cpu_sum = float(x.float().sum())
+            cpu_sum = float(torch.sparse.sum(x.float())) if x.is_sparse else float(
```

Comment:
This logic seems to be unrelated to fp16 reduce. I think it should be a separate PR and appropriate documented.